src/ui/components/post/post_listing.rs

```diff
@@ -23,18 +32,53 @@ pub fn PostListing<'a>(post_view: &'a PostView) -> impl IntoView {
 
   let post_state = RwSignal::new(post_view.clone());
 
+  let post_url = Memo::new(move |_| {
+    with!(|post_state| post_state
+      .post
+      .url
+      .as_deref()
+      .map(AsRef::as_ref)
+      .map(Rc::<str>::from))
+  });
+
+  let image_url = Memo::new(move |_| {
+    with!(
+      |post_state| post_state
+        .post
+        .thumbnail_url
+        .as_deref()
+        .map(AsRef::as_ref)
+        .map(Rc::from)
+        .or_else(|| post_url.get().filter(|url| is_image(url.as_ref()))) // Fall back to post url if no thumbnail, but only if it is an image url
+    )
+  });
```

Comment:
I had to use `Rc`s for these because they wouldn't play nice with the compiler otherwise.

Reply:
I'm trying to go through some of their logic here: 

- https://docs.rs/leptos/latest/leptos/fn.create_memo.html
- https://book.leptos.dev/reactivity/working_with_signals.html#making-signals-depend-on-each-other

Their main use case is for expensive computations, not simple reads, so I could see using one for the image_url (which has to do a regex check), but pry isn't necessary for the other simple reads. 

Also it seems like `create_memo(` is preferred over `Memo::new`

Reply:
> I'm trying to go through some of their logic here:
> https://docs.rs/leptos/latest/leptos/fn.create_memo.html
> https://book.leptos.dev/reactivity/working_with_signals.html#making-signals-depend-on-each-other
> Their main use case is for expensive computations, not simple reads, so I could see using one for the image_url (which has to do a regex check), but pry isn't necessary for the other simple reads.

In this case, I'm using `Memo` because if I use a regular derived signal, any change in `post_state` will cause reactive updates in every part of the UI that relies on parts of it regardless of if the relevant value has changed or not. Considering `PostView` is a large struct with many parts of the UI relying on only 1 or 2 of its properties, I think this choice is justified. [This section of the Leptos book talks about this behavior](https://book.leptos.dev/15_global_state.html?highlight=state#option-3-create-a-global-state-struct-and-slices):

> [B]ecause our whole state is wrapped in one signal, updating the value of one field will cause reactive updates in parts of the UI that only depend on the other... There’s a better way. You can take fine-grained, reactive slices by using [create_memo](https://docs.rs/leptos/latest/leptos/fn.create_memo.html) or [create_slice](https://docs.rs/leptos/latest/leptos/fn.create_slice.html) (which uses create_memo but also provides a setter). “Memoizing” a value means creating a new reactive value which will only update when it changes. “Memoizing a slice” means creating a new reactive value which will only update when some field of the state struct updates.

While that section is talking about global state for the entire app, the `post_state` is basically global state for a post listing.

> Also it seems like create_memo( is preferred over Memo::new

[The release for the most recent v0.7 beta suggests otherwise](https://github.com/leptos-rs/leptos/releases/tag/v0.7.0-beta):
> We're migrating away from create_ naming toward more idiomatic Rust naming patterns:
> - create_signal to signal (like channel)
> - create_rw_signal to RwSignal::new()
> - etc.
> - I've left some of the current functions in, marked deprecated; others may have been missed, but should be easy to find [via docs.rs](https://docs.rs/leptos/0.7.0-alpha/leptos).

Reply:
You're ahead of the docs, nice.